addons/mrp/report/report_stockpicking_operations.xml

```diff
@@ -0,0 +1,86 @@
+        </td>
+        <td align="right">
+            <span t-field="ml.quantity">3.00</span>
+            <span t-field="ml.product_uom_id" groups="uom.group_uom">units</span>
+            <span t-if="ml.move_id.product_packaging_id">
+                <span t-if="o.state != 'done'">
+                    (<span t-field="ml.product_packaging_qty" t-options='{"widget": "integer"}'/> <span t-field="ml.move_id.product_packaging_id.name"/>)
+                </span>
+                <span t-if="o.state == 'done'">
+                    (<span t-field="ml.product_packaging_qty" t-options='{"widget": "integer"}'/> <span t-field="ml.move_id.product_packaging_id.name"/>)
+                </span>
+            </span>
+        </td>
+        <td t-if="o.picking_type_id.code != 'incoming'" groups="stock.group_stock_multi_locations">
+            <span t-esc="ml.location_id.display_name">WH/Stock</span>
+                <t t-if="ml.package_id">
+                    <span t-field="ml.package_id">Package A</span>
+                </t>
+        </td>
+        <td t-if="o.picking_type_id.code != 'outgoing'" groups="stock.group_stock_multi_locations">
+            <div>
+                <span t-field="ml.location_dest_id">WH/Outgoing</span>
+                <t t-if="ml.result_package_id">
+                    <span t-field="ml.result_package_id">Package B</span>
+                </t>
+            </div>
+        </td>
+        <td class=" text-center h6" t-if="has_serial_number">
+            <span t-if="has_serial_number and (ml.lot_id or ml.lot_name)" t-esc="ml.lot_id.name or ml.lot_name" t-options="{'widget': 'barcode', 'humanreadable': 1, 'width': 400, 'height': 100, 'img_style': 'width:100%;height:35px;'}">
+                <div class="bg-light border-1 rounded d-flex flex-column align-items-center justify-content-center px-1 py-2 opacity-75 text-muted text-center">
+                    (serial barcode)
+                </div>
+            </span>
+        </td>
+        <td class="text-center" t-if="has_barcode">
+            <t t-if="product_barcode != ml.product_id.barcode">
+                <span t-if="ml.product_id and ml.product_id.barcode">
+                    <div t-field="ml.product_id.barcode" t-options="{'widget': 'barcode', 'symbology': 'auto', 'width': 400, 'height': 100, 'quiet': 0, 'img_style': 'height:35px;'}">
+                        <div class="bg-light border-1 rounded d-flex flex-column align-items-center justify-content-center px-1 py-2 opacity-75 text-muted text-center">
+                            (product barcode)
+                        </div>
+                    </div>
+                </span>
+                <t t-set="product_barcode" t-value="ml.product_id.barcode"/>
+            </t>
+        </td>
+    </template>
+    <template id="mrp_report_stock_picking" inherit_id="stock.report_picking">
+        <xpath expr="//table[@t-if='o.move_line_ids and o.move_ids_without_package']/tbody" position="replace">
```

Comment:
Don't replace whole section. Instead add proper hook to extend

Reply:
Hello @amoyaux,

I have replaced the entire <tbody> because I needed to update the table rows (<tr>) based on specific conditions and also to insert a row for the kit name. Given that the <tbody> currently only contains <tr> elements, I replaced it entirely rather than modifying individual rows. Additionally, I don't see a feasible way to add my conditions just before the <tr>, so I replaced the entire <body>.

Thank you.